src/indexer.c

```diff
@@ -355,7 +354,7 @@ RedisModuleDict *QueryIndex(RedisModuleCtx *ctx,
     for (int i = 0; i < predicate_count; i++) {
         if (index_predicate[i].type == NCONTAINS || index_predicate[i].type == NEQ ||
             index_predicate[i].type == LIST_NOTMATCH) {
-            result = QueryIndexPredicate(ctx, &index_predicate[i], result, shouldCreateNewDict);
+            result = QueryIndexPredicate(ctx, &index_predicate[i], result, 1);
```

Comment:
If we always call `QueryIndexPredicate` with `createResultDict == 1`, lets remove `createResultDict`.

Reply:
Agree. fixed in the latest commit.